### ISSUE TITLE
Output plugin should support multiple server addresses

### ISSUE THREAD

**Issue by driskell** [driskell/log-courier#129]
This is to keep track of future improvements to the output plugin.

- [ ] Change `addresses`/`port` parameters to `servers` for consistency with Log Courier itself
- [ ] Implement support for multiple servers with randomised initial and round robin failover

**Comment by msimerson** [comment]
This seems like a good place to also mention that a handy feature would be supporting different sets of of servers for different files. For example, I have logstash listening on port 5043 with `codec => plain` and another logstash listener on port 5044 with `codec => json`.

On the hosts generating the logs, I need to have two LSF/LC processes, one that ships syslog type files to the port 5043 "server" and another one that ships JSON encoded files to the port 5044 "server."

**Comment by driskell** [comment]
@msimerson IMO it would complicate matters, when you can simply receive everything as plain instead and then apply a JSON filter based on a conditional in Logstash to achieve the exact same result. It also centralises that configuration aspect too.

If there's a specific reason that's not feasible let me know though!

**Comment by msimerson** [comment]
> when you can simply receive everything as plain instead and then apply a JSON filter based on a conditional in Logstash

That's exactly the path I started down yesterday. More or less (from memory):

````
input {
  lumberjack {
    port => 5044
  }
}
filter {
  if [type] == "qph" {
    json => {
        source => "message"
    }
  }
}
````

That path was not fruitful and poking through the GH Issues revealed it wasn't just me and that path is actually more like a trail of tears. Within those issues, I got the general impression that such filtering options (particularly json and json_lines) were very problematic and the future for such would be this syntax (which is what I got working):

````
input {
  lumberjack {
    port => 5045
    codec => "json"
  }
}
````

**Comment by driskell** [comment]
Plugin is now in its own repository and I'm considering dropping the output plugin as there's lots of other options for Logstash to Logstash communication.

Codecs aren't supported by courier input still and those will be implemented when the 2.0 plugin is released
